include/glow/Backends/Backend.h

```diff
@@ -44,10 +44,10 @@ class Backend {
   virtual ~Backend() = default;
 
   /// Prepare the interpreter for execution of new code.
-  virtual void init() = 0;
+  virtual void init(std::unique_ptr<const IRFunction> IR) = 0;
```

Comment:
Overall, I like the direction.

Question: This `init` API states that this backend becomes the owner of the IR. Can we ever have a situation where we want to compile the same IR using different backends? If so, this unique_ptr may make it rather difficult. Of course, we could always clone the IR and provide each backend with its own copy as an option...

Reply:
The backends would want to do different things to the graph during lowering, right? It seems we would have to even clone the graph to use multiple backends.

Reply:
I could maybe see unit tests wanting to run the same IR with different backends...but it's telling that we don't have any now that do :).
I agree with @stoklund that any real scenario would probably do different lowering so we'd want to clone it.